### DISCUSSION TITLE
One model or Multiples Models for segmentation?

### DISCUSSION THREAD

**Discussion by AgustinSaturni** [Project-MONAI/tutorials#1774] [Q&A]
Hi everybody, Im developing a Model using Unet:
model = UNet(
    spatial_dims=3,
    in_channels=1,
    out_channels=8, 
    channels=(16, 32, 64, 128, 256),
    strides=(2, 2, 2, 2),
    num_res_units=2,
    norm=Norm.BATCH,
).to(device)
In this case I have 7 bones to segment and I would like to know what do you think is best for this situation. I dont know if its convinient to train one model to segment every bone OR train multiples models wich each model can detect each bone.
Wich option is better??
Thank you!!

**Comment by KumoLiu** [comment]
Hi @AgustinSaturni, you can start by trying a single model approach. It's simpler and can often be surprisingly effective. You can then evaluate its performance on each bone, if the performance is acceptable for all bones, this is the most efficient solution. If you notice that certain bones are poorly segmented, you could consider training additional models specifically for those bones or applying techniques like multi-task learning.

Thanks.

**Reply by AgustinSaturni** [reply]
Hi @KumoLiu  thank you for replying. Im gonna try this approach. By the way, I noticed that the full-body CT segmentation model in MONAI is trained with SegResNet instead of Unet. Do you know if it's better than Unet when you want to segment multiples things or what are the pros that this architecture has?
Thank you

**Reply by diazandr3s** [reply]
> Hi @KumoLiu thank you for replying. Im gonna try this approach. By the way, I noticed that the full-body CT segmentation model in MONAI is trained with SegResNet instead of Unet. Do you know if it's better than Unet when you want to segment multiples things or what are the pros that this architecture has? Thank you

You can find more information about this here: https://arxiv.org/abs/1810.11654
